neuralpp/symbolic/sympy_expression.py

```diff
@@ -329,22 +335,28 @@ def native_arguments(self) -> Tuple[sympy.Basic, ...]:
 
     @staticmethod
     def from_sympy_function_and_general_arguments(sympy_function: sympy.Basic, arguments: List[Expression]) -> \
-            SymPyFunctionApplication:
+            SymPyExpression:
         sympy_arguments = [SymPyExpression._convert(argument) for argument in arguments]
         type_dict = _build_type_dict_from_sympy_arguments(sympy_arguments)
 
         # Stop evaluation, otherwise Add(1,1) will be 2 in sympy.
         if sympy_function == sympy.Min or sympy_function == sympy.Max:
             # see test/sympy_test.py: test_sympy_bug()
             sympy_object = sympy_function(*[sympy_argument.sympy_object for sympy_argument in sympy_arguments],
-                                          evaluate=False)
+                                          evaluate=global_parameters.sympy_evaluate)
         elif sympy_function == sympy.Piecewise:
             sympy_object = sympy_piecewise_from_if_then_else(
                 *[sympy_argument.sympy_object for sympy_argument in sympy_arguments])
         else:
-            with sympy.evaluate(False):
+            with sympy.evaluate(global_parameters.sympy_evaluate):
```

Comment:
Instead of always open the context with `sympy.evaluate(False)`, we set the value to our global parameter `global_parameters.sympy_evaluate`, which if call by `simplify()`, the value is set to True to allow evaluation.